### ISSUE TITLE
install dependencies of probfit numpy, cython, pyqt

### ISSUE THREAD

**Issue by deisi** [scikit-hep/probfit#79]
Hey I just installed probfit (#24e19f0) in a virtualenv without system-site-packages. 

## Numpy is needed pre installation
To install probfit at all, one needs to preinstall numpy manually with:
`pip install numpy`

## Cython needed even for not developer build
Import probfit with `import probfit` I get: 
``` python
ImportError                               Traceback (most recent call last)
<ipython-input-1-05ae4a7acf03> in <module>()
----> 1 import probfit

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/probfit/__init__.py
in <module>()
      6 """
      7
----> 8 from .costfunc import UnbinnedLH, BinnedLH, Chi2Regression,
BinnedChi2, SimultaneousFit
      9 from .pdf import doublegaussian, ugaussian, gaussian, crystalball, \
     10     argus, cruijff, linear, poly2, poly3, novosibirsk, \

ImportError: No module named 'probfit.costfunc'
```

After manually installing cython with:
`pip install cython`

and reinstallation of probfit with:
`pip uninstall probfit`
`pip install .`

## pyqt needed
Now it shows 
```python
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-05ae4a7acf03> in <module>()
----> 1 import probfit

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/probfit/__init__.py
in <module>()
      6 """
      7
----> 8 from .costfunc import UnbinnedLH, BinnedLH, Chi2Regression,
BinnedChi2, SimultaneousFit
      9 from .pdf import doublegaussian, ugaussian, gaussian, crystalball, \
     10     argus, cruijff, linear, poly2, poly3, novosibirsk, \

/home/malte/Projects/probfit/probfit/costfunc.pyx in init
probfit.costfunc (probfit/costfunc.c:16756)()
      4 cimport numpy as np
      5 from libc.math cimport exp, pow, fabs, log, tgamma, lgamma, sqrt
----> 6 from matplotlib import pyplot as plt
      7 from . import plotting
      8 from ._libstat cimport compute_nll, compute_chi2_f,
compute_bin_chi2_f, \

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/pyplot.py
in <module>()
    112
    113 from matplotlib.backends import pylab_setup
--> 114 _backend_mod, new_figure_manager, draw_if_interactive, _show =
pylab_setup()
    115
    116 _IP_REGISTERED = None

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/backends/__init__.py
in pylab_setup()
     30     # imports. 0 means only perform absolute imports.
     31     backend_mod = __import__(backend_name,
---> 32                              globals(),locals(),[backend_name],0)
     33
     34     # Things we pull in from all backends

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/backends/backend_qt4agg.py
in <module>()
     16
     17
---> 18 from .backend_qt5agg import FigureCanvasQTAggBase as
_FigureCanvasQTAggBase
     19
     20 from .backend_agg import FigureCanvasAgg

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/backends/backend_qt5agg.py
in <module>()
     14
     15 from .backend_agg import FigureCanvasAgg
---> 16 from .backend_qt5 import QtCore
     17 from .backend_qt5 import QtGui
     18 from .backend_qt5 import FigureManagerQT

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/backends/backend_qt5.py
in <module>()
     29     figureoptions = None
     30
---> 31 from .qt_compat import QtCore, QtGui, QtWidgets,
_getSaveFileName, __version__
     32 from matplotlib.backends.qt_editor.formsubplottool import
UiSubplotTool
     33

/home/malte/.virtualenvs/probfittest/lib/python3.5/site-packages/matplotlib/backends/qt_compat.py
in <module>()
    135     # have been changed in the above if block
    136     if QT_API in [QT_API_PYQT, QT_API_PYQTv2]:  # PyQt4 API
--> 137         from PyQt4 import QtCore, QtGui
    138
    139         try:

ImportError: No module named 'PyQt4'
```

So it also needs pyqt, PyQt4 from pip doesnt install for me and I installed it for the system with:
`pacman -S python-pyqt4`
(I wonder why this works, because it should but it does)

And finally probfit works :)

**Comment by alexpearce** [comment]
I'm sorry for the trouble. probfit has been in a period of flux recently, particularly with regards to packaging.

If you want to install the package from the repository, you need Cython in order to compile the Cython source files to C files (the latter of which used to be included in the repo). You also need numpy for this step, as well as to compile the C files. If you install from the git repo, probfit considers that you are performing a "developer build". In the tagged releases, you won't need Cython as the C files will be included.

The PyQt dependency problem is confusing. Perhaps your matplotlibrc specifies the Qt backend as the default?

**Comment by henryiii** [comment]
This was fixed in #105
